doc/companies.md

```diff
@@ -1 +1,881 @@
 # Companies Module
+Module | HTTP Method | URL | Description
+--- | --- | --- | ---
+[Add](#add) | POST | /companies | Add Data Companies
+[View](#view) | GET | /companies/:id | View Data Companies
+[Edit](#edit) | PUT | /companies/:id | Edit Data Companies
+[Delete](#delete) | DELETE | /companies/:id | Delete Data Companies
+
+## <a name="add"></a>Add Data Companies
+
+### Endpoint
+POST /companies
+
+### Database
+![](./Companies.png)
+
+### Headers
+Key | Value
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+company_code | string | TC  
+name | string | Test Company
+address | string | JL. Test Company
+npwp | string | 123
+nomor_kukuh | string | 123
+tgl_kukuh | date | 12
+phone_num | string | 021  
+fax_num | string | 021
+seri_pajak | string | 12
+kota | string | JAKARTA
+```
+{
+    "company_code": "TC",
+    "name": "Test Company",
+    "address": "JL. Test Company",
+    "npwp": "123",
+    "nomor_kukuh": "123",
+    "tgl_kukuh": "12",
+    "phone_num": "021",
+    "fax_num": "021",
+    "seri_pajak": "12",
+    "kota": "JAKARTA"
+}
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+500 | Internal Server Error | some un-handle error in server
+201 | Created | Created
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-201",
+    "status_message": "Created",
+    "data": {
+        "id":"5e8d6470-f654-24c8-93ac-332e6c3dra2e",
+        "company_code": "TC",
+        "name": "Test Company",
+        "address": "JL. Test Company",
+        "npwp": "123",
+        "nomor_kukuh": "123",
+        "tgl_kukuh": "12",
+        "phone_num": "021",
+        "fax_num": "021",
+        "seri_pajak": "12",
+        "kota": "JAKARTA"
+    }
+}
+```
+
+### Logic
+
+#### Validation
+- company_code : unique and not empty
```

Comment:
di screnario test, ada company_code is required, mungkin bisa ditambahkan di baris ini